File: go/restclient/restClient.go
package restclient

import (
	"fmt"
	"net/http"
	"sync"
	"time"
)

var testCounts = []int{100, 200, 300}

const baseURL = "http://localhost:5000"

var endpoints = map[string]string{
	"usuarios":  "/usuarios",
	"musicas":   "/musicas",
	"playlists": "/playlists",
}

func testLoad(endpoint, name string, numTest int) {
	url := baseURL + endpoint
	start := time.Now()

	var wg sync.WaitGroup

	for i := 0; i < numTest; i++ {
		wg.Add(1)
		func(i int) {
			defer wg.Done()
			resp, err := http.Get(url)
			if err != nil {
				fmt.Printf("Request %d failed: %v\n", i, err)
				return
			}
			defer resp.Body.Close()
			if resp.StatusCode != http.StatusOK {
				fmt.Printf("Request %d failed with status: %s\n", i, resp.Status)
			}
		}(i)
	}
	wg.Wait()

	end := time.Now()
	fmt.Printf("%s : %v seconds\n", name, end.Sub(start).Seconds())
}

func RunTests() {
	for _, numTest := range testCounts {
		fmt.Printf("%d requests:\n", numTest)
		testLoad(endpoints["usuarios"], "Usuarios", numTest)
		testLoad(endpoints["musicas"], "Musicas", numTest)
		testLoad(endpoints["playlists"], "Playlists", numTest)
	}
}

File: csharp/RestClient.cs
using System;
using System.Net.Http;
using System.Threading.Tasks;

public class RestClient
{
    private readonly HttpClient _client;

    public RestClient(string baseAddress)
    {
        _client = new HttpClient { BaseAddress = new Uri(baseAddress) };
    }

    private async Task SendRequestAsync(string endpoint, int i)
    {
        try
        {
            var response = await _client.GetAsync(endpoint);
            response.EnsureSuccessStatusCode();
        }
        catch (Exception ex)
        {
            Console.WriteLine($"Request {i} to {endpoint} failed: {ex.Message}");
        }
    }

    public async Task RunTests()
    {
        Console.WriteLine("Running REST tests...");

        var endpoints = new[] { "usuarios", "musicas", "playlists" };
        var numTests = new[] { 100, 200, 300 };

        foreach (var numTest in numTests)
        {
            var start = DateTime.Now;
            Console.WriteLine($"Running tests with {numTest} requests:");

            foreach (var endpoint in endpoints)
            {
                for (int i = 0; i < numTest; i++)
                {
                    await SendRequestAsync(endpoint, i);
                }
            }

            var end = DateTime.Now;
            Console.WriteLine(
                $"REST tests with {numTest} requests completed in {(end - start).TotalSeconds} seconds"
            );
        }
    }
}
